src/monmove.cpp

```diff
@@ -940,7 +978,7 @@ bool monster::bash_at( const tripoint &p )
         return false;
     }
     bool try_bash = !can_move_to( p ) || one_in( 3 );
-    bool can_bash = g->m.is_bashable( p ) && ( has_flag( MF_BASHES ) || has_flag( MF_BORES ) );
+    bool can_bash = g->m.is_bashable( p ) && bash_skill() > 0;
```

Comment:
You might want a two parameter version of `is_bashable` with the second paramter `const Creature &`?